### PR TITLE
Update CHANGELOG.md

### PR DESCRIPTION
**Pull request by SwRaw**
Updated changelog as per doc standards

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,74 +1,83 @@
-# Change Log for Tensile
+# Changelog for Tensile
 
 ## (Unreleased) Tensile 4.42.0
```
Comment (SwRaw) [inline CHANGELOG.md:3]: Can I replace this with Tensile 4.42.0 for ROCm 6.3?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,74 +1,83 @@
...
+- Contributor and developer guide
+- Testing and documentation for `MasterSolutionLibrary.ArchitectureIndexMap` and `remapSolutionIndicesStartingFrom`
+- gfx12 support
+- Functions for writing master file
+- `tPrint` and reconcile printing options
+- Python unit test coverage report
+- Factor embed library logic into function and test
+- `clang++` as cxx-compiler option for Windows
+- Logic to cope with different compilers
+- `generateManifest` function and rename generateManifest to toFile and move to Utilities
```
Comment (SwRaw) [inline CHANGELOG.md:16]: @bstefanuk If we have renamed generateManifest function to `toFile` already then why are we mentioning that we have added generateManifest function? Shouldn't we write "Added toFile function in Utilities".

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,74 +1,83 @@
...
+- Refactored `generateLogicDataAndSolutions`
+- Restricted XCC mapping to gfx942
+- Refactored argument parsing in `TensileCreateLibrary`
+- Disabled failing rhel9 tests
+- Changed line length to 100 characters for formatting
+- Changed YAML operations to use C `libyaml` backend
+- Improved warning text
+- Updated clang support for Windows
+- Updated `supportedCompiler` function
 - use conditional choices and defaults
```
Comment (SwRaw) [inline CHANGELOG.md:45]: @bstefanuk  What exactly is changing here?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,74 +1,83 @@
...
-- remove duplicate which function and minor cleanup
-- refactor sanity check in TensileCreateLibrary
-- factor client config logic from TensileCreateLibrary main into createClientConfig
+- Refactored sanity check in `TensileCreateLibrary`
+- Moved client config logic from `TensileCreateLibrary` main into `createClientConfig`
 - use glob to find logic files in TensileCreateLibrary
 - use function to confirm supported compiler rather than raw logic
```
Comment (SwRaw) [inline CHANGELOG.md:49]: @bstefanuk What exactly is changing here? It seems like this should be listed under "Added".

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,74 +1,83 @@
...
-- remap XCC-based workgroup for Stream-K kernels
-- add two-tile algorithm with Stream-K after DP
-- add atomic 2-tile Stream-K and clean-up tuning parameters
+- Contributor and developer guide
+- Testing and documentation for `MasterSolutionLibrary.ArchitectureIndexMap` and `remapSolutionIndicesStartingFrom`
+- gfx12 support
+- Functions for writing master file
+- `tPrint` and reconcile printing options
+- Python unit test coverage report
+- Factor embed library logic into function and test
```
Comment (SwRaw) [inline CHANGELOG.md:13]: @bstefanuk Factor embed logic or added embed logic?

Comment (bstefanuk) [reply CHANGELOG.md:13]: Please keep the term "factor" as it is technically more correct.

File: CHANGELOG.md (hunk 2 above)
Comment (bstefanuk) [reply CHANGELOG.md:16]: This update is fine, but please mention that the generate manifest step has been moved into this function as well.

File: CHANGELOG.md (hunk 3 above)
Comment (bstefanuk) [reply CHANGELOG.md:45]: Please change this to "Use conditional choices and defaults for clang support on Windows"

File: CHANGELOG.md (hunk 4 above)
Comment (bstefanuk) [reply CHANGELOG.md:49]: A function was implemented to encapsulate logic.

Sometimes the difference between a change and addition is fuzzy, you can move this into "Added" if you please.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,74 +1,83 @@
...
+- Deprecated flag from CI profiling job
+- Diagnostic print
+- Globals from `prepAsm`
+- Deprecated `package-library` option
+- Duplicate `which` function and minor cleanup
+
+### Optimized
+
+- Use analytical grid size prediction model for Stream-K
+- Remapped XCC-based workgroup for Stream-K kernels
```
[the remark this answers is not in the document]
Comment (bstefanuk) [reply CHANGELOG.md:66]: I have no preference on this section, any thoughts @babakpst or @AlexBrownAMD?

[the remark this answers is not in the document]
Comment (AlexBrownAMD) [reply CHANGELOG.md:66]: No real preference on where they belong, probably makes the most sense under either Optimized or Added. To clarify, these were new features added to optimize the performance of stream-k kernels.

File: CHANGELOG.md (hunk 2 above)
Comment (SwRaw) [reply CHANGELOG.md:16]: done.

File: CHANGELOG.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (SwRaw) [reply CHANGELOG.md:66]: Done.

File: CHANGELOG.md (hunk 3 above)
Comment (SwRaw) [reply CHANGELOG.md:45]: Done

File: CHANGELOG.md (hunk 5 above)
Comment (SwRaw) [reply CHANGELOG.md:13]: Ok

**Review by bstefanuk — APPROVED** [review]

**Pull request merged**